exporter/collector/monitoredresource.go

```diff
@@ -0,0 +1,166 @@
+			podName:       {semconv.AttributeK8SPodName},
+		},
+		k8sNode: {
+			location:    {semconv.AttributeCloudAvailabilityZone},
+			clusterName: {semconv.AttributeK8SClusterName},
+			nodeName:    {semconv.AttributeK8SNodeName},
+		},
+		k8sCluster: {
+			location:    {semconv.AttributeCloudAvailabilityZone},
+			clusterName: {semconv.AttributeK8SClusterName},
+		},
+		awsEc2Instance: {
+			instanceID: {semconv.AttributeHostID},
+			region:     {semconv.AttributeCloudAvailabilityZone},
+			awsAccount: {semconv.AttributeCloudAccountID},
+		},
+		genericTask: {
+			location:  {semconv.AttributeCloudAvailabilityZone, semconv.AttributeCloudRegion},
+			namespace: {semconv.AttributeServiceNamespace},
+			job:       {semconv.AttributeServiceName},
+			taskID:    {semconv.AttributeServiceInstanceID},
+		},
+		genericNode: {
+			location:  {semconv.AttributeCloudAvailabilityZone, semconv.AttributeCloudRegion},
+			namespace: {semconv.AttributeServiceNamespace},
+			nodeID:    {semconv.AttributeHostID, semconv.AttributeHostName},
+		},
+	}
+)
+
+// Transforms pdata Resource to a GCM Monitored Resource. Any resource attributes not accounted
+// for in the monitored resource which should be merged into metric labels are also returned.
+func (m *metricMapper) resourceMetricsToMonitoredResource(
+	resource pdata.Resource,
+) (*monitoredrespb.MonitoredResource, labels) {
+	attrs := resource.Attributes()
+	cloudPlatform := getStringOrEmpty(attrs, semconv.AttributeCloudPlatform)
+
+	switch cloudPlatform {
+	case semconv.AttributeCloudPlatformGCPComputeEngine:
+		return createMonitoredResource(gceInstance, attrs)
+	case semconv.AttributeCloudPlatformGCPKubernetesEngine:
+		// Try for most to least specific k8s_container, k8s_pod, etc
+		if _, ok := attrs.Get(semconv.AttributeK8SContainerName); ok {
+			return createMonitoredResource(k8sContainer, attrs)
+		} else if _, ok := attrs.Get(semconv.AttributeK8SPodName); ok {
+			return createMonitoredResource(k8sPod, attrs)
+		} else if _, ok := attrs.Get(semconv.AttributeK8SNodeName); ok {
+			return createMonitoredResource(k8sNode, attrs)
+		}
+		return createMonitoredResource(k8sCluster, attrs)
+	case semconv.AttributeCloudPlatformAWSEC2:
+		return createMonitoredResource(awsEc2Instance, attrs)
+	default:
+		// Fallback to generic_task
+		_, hasServiceName := attrs.Get(semconv.AttributeServiceName)
+		_, hasServiceInstanceID := attrs.Get(semconv.AttributeServiceInstanceID)
+		if hasServiceName && hasServiceInstanceID {
+			return createMonitoredResource(genericTask, attrs)
+		}
+
+		// If not possible, fallback to generic_node
+		return createMonitoredResource(genericNode, attrs)
+	}
+}
+
+func createMonitoredResource(
+	monitoredResourceType string,
+	resourceAttrs pdata.AttributeMap,
+) (*monitoredrespb.MonitoredResource, labels) {
+	mappings := monitoredResourceMappings[monitoredResourceType]
+	mrLabels := make(map[string]string, len(mappings))
+	// TODO handle extra labels
```

Comment:
Is there a task/bug to handle this?

Reply:
Created a bug for myself. I can do it another PR